### PR TITLE
Gem3 mapper

### PR DESCRIPTION
**Pull request by karl616**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by karl616** [comment]
This is gem3-mapper: https://github.com/smarco/gem3-mapper
I think it would be a worthy addition to bioconda

File: recipes/gem3-mapper/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+#adding include and library paths in order to find bzip2
+sed -i.bak '
+    /^PATH_INCLUDE=/ s@$@ -I'$PREFIX'/include@
+    /^PATH_LIB=/ s@$@ -L'$PREFIX'/lib@
+  ' Makefile.mk.in
+
+
+./configure
```
Comment (bgruening) [inline recipes/gem3-mapper/build.sh]: ./configure --prefix=$PREFIX?

File: recipes/gem3-mapper/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  host:
+    - bzip2
+    - zlib
+  run:
+    - bzip2
+    - zlib
+
+test:
+  commands:
+          #    - gem-mappability -h 2>&1 |grep -e "-I <index_prefix>.*(mandatory)"
```
Comment (bgruening) [inline recipes/gem3-mapper/meta.yaml]: anything wrong with this test?

File: recipes/gem3-mapper/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    - gem-mapper 2>&1 | grep 'index|-I <index_file.gem>'
+    - gem-indexer -h 2>&1 | grep 'input|-i <input_file> (Multi-FASTA)'
+    - gem-retriever -h 2>&1 | grep 'index|-I <index_file.gem> (GEM archive)'
+
+about:
+  home: https://github.com/smarco/gem3-mapper
+  license: GPL3
```
Comment (bgruening) [inline recipes/gem3-mapper/meta.yaml]: license_file: LICENSE could be added.
Can you maybe also include a DOI or bio.tools ID?

File: recipes/gem3-mapper/build.sh (hunk 1 above)
Comment (karl616) [reply recipes/gem3-mapper/build.sh]: Nope, it is not enough. I also tried with CFLAGS and LDFLAGS, but they are not respected. Thanks for the suggestion though, I didn't think of it earlier.

File: recipes/gem3-mapper/meta.yaml (hunk 3 above)
Comment (karl616) [reply recipes/gem3-mapper/meta.yaml]: Will do.
I'm adding the original paper. There is an entry in bio.tools/gemmapper, but it is currently pointing to an older version, not updated, version of the software. I will add it for now and ask the developer to try and update it.

File: recipes/gem3-mapper/meta.yaml (hunk 2 above)
Comment (karl616) [reply recipes/gem3-mapper/meta.yaml]: the particular executable is not included in this version. I started with an older version, but changed along the way. I will remove it.

**Comment by karl616** [comment]
@bgruening Thanks for the feedback, always appreciated

**Pull request merged**